crates/burn-import/src/burn/node/gather.rs

```diff
@@ -27,7 +27,15 @@ impl<PS: PrecisionSettings> NodeCodegen<PS> for GatherNode {
         node_position: usize,
     ) -> proc_macro2::TokenStream {
         let dim = self.dim.to_tokens();
-        let input = scope.tensor_use_owned(&self.input, node_position);
+        let input = match &self.input {
+            Type::Tensor(in_tensor) => scope.tensor_use_owned(in_tensor, node_position),
+            Type::Shape(in_shape) => {
+                // TODO: do we need a shape_use_owned for the scope, in case the Shape is used multiple times?
```

Comment:
We could probably pass a reference (make it slice) to `Tensor::from_data` since there is an From implementation. This way we borrow and underlying elements are copied:

<img width="864" alt="image" src="https://github.com/user-attachments/assets/a59593b3-f911-497b-8cae-e4f85ef584b7">

@laggui what do you think?

Reply:
Agreed! 👍

Reply:
Done as suggested